src/components/screens/editor/editor-screen.tsx

```diff
@@ -502,6 +474,107 @@ const EditorScreen: React.FC<Props> = ({
     }
 
     setEditedTrainingMaterial({ ...editedTrainingMaterial, [name]: value });
+    setDataChanged(true);
+  }
+
+  /**
+   * Event handler for save button click
+   */
+  const onSaveClick = async () => {
+    if (!accessToken || !storyData.story) {
+      return;
+    }
+    try {
+      const knotsApi = Api.getKnotsApi(accessToken);
+      const intentsApi = Api.getIntentsApi(accessToken);
+      const trainingMaterialsApi = Api.getTrainingMaterialApi(accessToken);
+      let currentStory = story;
+      if (story) {
+        currentStory = await Api.getStoriesApi(accessToken).updateStory({ storyId: storyId, story: story });  
+      }
+
+      let currentKnots = knots ? knots : [];
```

Comment:
This might be quite slow on larged stories, all update operations should be done within same Promise.all to ensure they are executed asynchronously